### PR TITLE
test: speed up commitment-tree Sinsemilla tests in CI

### PR DESCRIPTION
**Pull request by QuantumExplorer**
## Problem

`commitment_tree::tests::storage_tests::append_many_raw_byte_for_byte_matches_per_leaf` took **4+ minutes** in CI (nextest flagged it SLOW at 60/120/180/240s). Its per-leaf reference path calls `append_raw` once per leaf, and each append does an eager depth-32 Sinsemilla root walk — Pallas-curve arithmetic — for N up to 10,000, in an unoptimized debug build with llvm-cov instrumentation on top.

## Changes

1. **Optimize the curve crates in dev/test builds**: added `[profile.dev.package.*] opt-level = 3` to the workspace `Cargo.toml` for `pasta_curves`, `halo2_gadgets`, `halo2_proofs`, `sinsemilla`, `orchard`, and `incrementalmerkletree`. Workspace crates stay unoptimized and debuggable; release builds are unaffected. This is a ~20–50× speedup for everything touching Sinsemilla.

2. **Split the equivalence test**: N ∈ {0, 1, 2, 3, 100} stays in the default suite (~0.5s); N ∈ {2048, 10,000} moved to `append_many_raw_byte_for_byte_matches_per_leaf_large` marked `#[ignore]` (~46s even with optimized deps). The existing **Slow Tests** workflow already runs `--ignored` tests for this crate on commitment-tree PRs, so the large sizes keep CI coverage — same precedent as the other Sinsemilla-heavy ignored tests.

3. **Un-ignored** `test_roundtrip_with_many_leaves` and `test_serialize_roundtrip_with_many_leaves` (previously `#[ignore]` with "~60s" comments): with the profile change they run in under a second, so they're back in the default suite.

## Result

Full `grovedb-commitment-tree` suite: **4+ minutes → ~4 seconds** (58 passed, 1 ignored → covered by Slow Tests workflow).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Tests**
  * Enabled additional commitment frontier and commitment tree round-trip tests in the standard test suite.
  * Added broader coverage for batched append equivalence, while retaining a separate extended test for larger scenarios.

* **Performance**
  * Improved development and test build performance for selected commitment-tree components.

### CODE REVIEW

File: grovedb-commitment-tree/src/commitment_tree/tests.rs [hunk 1]
```diff
@@ -1248,6 +1242,24 @@ mod storage_tests {
         }
     }
 
+    /// Fast sizes: 0 / 1 / 2 / 3 cover edge cases around the very-empty and
+    /// pre-compaction shapes; 100 spans the buffer mid-range. The per-leaf
+    /// reference path is O(N) depth-32 Sinsemilla walks, so larger sizes live
+    /// in the `#[ignore]`d companion test below.
```
[the remark this answers is not in the document]
Comment (QuantumExplorer) [reply grovedb-commitment-tree/src/commitment_tree/tests.rs]: This is Claude. Fixed in 677ea181 — you're right: with TEST_CHUNK_POWER=1 the epoch size is 2, so 2048 entries span 1024 epochs and N=2/3 are the actual epoch-boundary edge cases. Both doc comments now describe the real shapes. (The wrong rationale predates this PR; it was carried over when splitting the test.)

**Comment by QuantumExplorer** [comment]
This is Claude. Re the nitpick about capturing `CostContext` / `CommitmentAppendResult` in `assert_append_many_matches_per_leaf`: declining, for two reasons.

1. **Cost equality between the two paths would be a wrong assertion by design.** The batched path is intentionally cheaper — it skips the per-leaf depth-32 root walk and recovers it once at the end. That cost contract is exactly what the adjacent `append_no_root_cost_omits_per_leaf_depth_walk` test covers, including the structural relationship that justifies the difference.

2. **Final-state equality is already asserted byte-for-byte.** The helper compares the Sinsemilla root, the full `CommitmentFrontier` serialization, the BulkAppendTree state root, and the total count — which together fully determine the observable tree state. A regression in positions or compaction state would change the frontier bytes or the state root and fail the existing assertions.

Note this PR only splits/relocates the pre-existing test for CI speed; its assertion surface is unchanged from develop.
